### DISCUSSION TITLE
Division by zero

### DISCUSSION THREAD

**Discussion by wllmlgn** [gephi/gephi#2624] [Show and tell]
Hi,

I love using Gephi - mostly. It helped me plot out my first novel. Preparing to write the second one, however, caused a lot of bad memories to come back.

There's this issue where the graph completely disappears because the transform algorithm didn't like a node or arc.

The second time I got the source and ran it through the debugger I found a severe issue. It's in ForceAtlasLayout.java, when the getCooling() function is called, returns zero, and is then used as a divisor. Since no exception is thrown, all the nodes have an infinite distance from each other, and the graph disappears. Two-line, uncommitted, workaround
                    double cooling = getCooling ();
                    if (cooling == 0) cooling = 1;

That's when I stopped looking, but obviously, there is a deeper issue with the value of cooling.

Thanks
WF Logan

**Comment by mbastian** [comment]
@jacomyma Can you take a look at this one? Thx
